core/src/test/kotlin/fr/sncf/osrd/pathfinding/constraints/SignalingSystemConstraintsTest.kt

```diff
@@ -0,0 +1,68 @@
+package fr.sncf.osrd.pathfinding.constraints
+
+import fr.sncf.osrd.api.pathfinding.constraints.SignalingSystemConstraints
+import fr.sncf.osrd.graph.Pathfinding
+import fr.sncf.osrd.sim_infra.api.Block
+import fr.sncf.osrd.sim_infra.api.BlockId
+import fr.sncf.osrd.sim_infra.api.TrackChunk
+import fr.sncf.osrd.train.RollingStock
+import fr.sncf.osrd.train.TestTrains
+import fr.sncf.osrd.utils.DummyInfra
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.io.IOException
+import java.net.URISyntaxException
+import java.util.stream.Stream
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SignalingSystemConstraintsTest {
+    private var signalingSystemConstraints: SignalingSystemConstraints? = null
+    private var infra: DummyInfra? = null
+
+    @BeforeAll
+    @Throws(IOException::class, URISyntaxException::class)
+    fun setUp() {
+        /*       c1
+                ^  \
+               /    v
+        a --> b     d --> e
+               \    ^
+                v  /
+                 c2
+         */
```

Comment:
The infra topology doesn't matter here, you can just instantiate one route for each type in a linear way.

You don't even need to keep the infra itself, you just need the block ids.

It would be more straightforward to move the setup directly in the single test here.

Reply:
done